### DISCUSSION TITLE
Wall Temperature Distribution

### DISCUSSION THREAD

**Discussion by appheadspace** [firemodels/fds#13708] [Q&A] [answered]
Hi, Please help me in implementing temperature distribution of wall in 2D cavity. Kindly help, I'm new to FDS and after seeing FDS user guide, I've tried 3 approaches and all are not working !

I've tried this:

APPROACH-1:::

&SURF ID='bottom_wall', RAMP_T='bottom_temp_ramp' /
&RAMP ID='bottom_temp_ramp', T=0.0, F=50.0 /
&RAMP ID='bottom_temp_ramp', T=0.00067, F=46.91 /
&RAMP ID='bottom_temp_ramp', T=0.0133, F=45.15 /

Approach-2:::

&OBST XB=0,0.750,-0.00005,0.00005,0,0 , SURF_ID='bottom_wall' /
&SURF ID = 'bottom_wall', COLOR = 'GRAY', RAMP_T = Bottom_Temp_Ramp, TMP_FRONT = 30/

&RAMP ID='Bottom_Temp_Ramp', T=0, F=1.6667
&RAMP ID='Bottom_Temp_Ramp', T=0.00015, F=1.5987
&RAMP ID='Bottom_Temp_Ramp', T=0.005025, F=1.5777

Approach -3:::

&OBST XB=0,0.750,-0.00005,0.00005,0,0 , SURF_ID='bottom_wall' /
&SURF ID = 'bottom_wall', COLOR = 'GRAY'/

&OBST XB =0,0,-0.00005,0.00005,0,0 ,SURF_ID='BW1'/
&SURF ID = 'BW1',TMP_FRONT =50/

&OBST XB =0.00015,0.00015,-0.00005,0.00005,0,0 ,SURF_ID='BW2'/
&SURF ID = 'BW2',TMP_FRONT =47.96/

&OBST XB =0.005025,0.005025,-0.00005,0.00005,0,0 ,SURF_ID='BW3'/
&SURF ID = 'BW3',TMP_FRONT =47.33/

bottom wall temperatures are showing 0 (for approach-1) and 30(for approach 2,3) at all locations. 
(PS: In my case INIT temp and TMPA , both are 30)

Kindly guide me. I'm new to FDS.

**Comment by mcgratta** [comment]
```
TMP(T) = TMPA + F*(TMP_FRONT-TMPA)
```

**Reply by appheadspace** [reply]
Thanks @mcgratta for your reply!. I tried the below approach by changing TMP_FRONT  but the temp is 13.369912 for all locations in bottom wall. Please check and Kindly guide me.
========

&MISC STRATIFICATION=F, SIMULATION_MODE='LES', TMPA = 30./

&SURF ID = 'bottom_wall', 	COLOR = 'GRAY',  RAMP_T = Bottom_Temp_Ramp, TMP_FRONT = 20/

&RAMP ID='Bottom_Temp_Ramp',  T=0, F=-2/
&RAMP ID='Bottom_Temp_Ramp',  T=0.00015, F=-1.674/
&RAMP ID='Bottom_Temp_Ramp',  T=0.005025, F=-1.615/
&RAMP ID='Bottom_Temp_Ramp',  T=0.009975, F=-1.545/
&RAMP ID='Bottom_Temp_Ramp',  T=0.020025, F=-1.456/
&RAMP ID='Bottom_Temp_Ramp',  T=0.039975, F=-1.093/
&RAMP ID='Bottom_Temp_Ramp',  T=0.075, F=-0.643/
&RAMP ID='Bottom_Temp_Ramp',  T=0.15, F=-0.105/
&RAMP ID='Bottom_Temp_Ramp',  T=0.225, F=0.192/
&RAMP ID='Bottom_Temp_Ramp',  T=0.3, F=0.416/
&RAMP ID='Bottom_Temp_Ramp',  T=0.375, F=0.599/
&RAMP ID='Bottom_Temp_Ramp',  T=0.45, F=0.754/
&RAMP ID='Bottom_Temp_Ramp',  T=0.525, F=0.911/
&RAMP ID='Bottom_Temp_Ramp',  T=0.6, F=1.184/
&RAMP ID='Bottom_Temp_Ramp',  T=0.675, F=1.404/
&RAMP ID='Bottom_Temp_Ramp',  T=0.729975, F=1.822/
&RAMP ID='Bottom_Temp_Ramp',  T=0.740025, F=1.87/
&RAMP ID='Bottom_Temp_Ramp',  T=0.7485, F=1.926/
&RAMP ID='Bottom_Temp_Ramp',  T=0.75, F=2/

**Comment by mcgratta** [comment]
Post a very simple input file.

**Reply by appheadspace** [reply]
Thanks for ur reply! PFA input file and line file. 

[chid.txt](https://github.com/user-attachments/files/17652338/chid.txt)
[chid_line.csv](https://github.com/user-attachments/files/17652339/chid_line.csv)

**Reply by drjfloyd** [reply]
This identical question is also being discussed in 13704.  Please limit your discussion a specific issue to a single thread.

**Reply by appheadspace** [reply]
Hi @drjfloyd , That discussion was showing "closed". I don't know why and also there was no option to re-open it. That's why created a new one. However, I've noted your point and I will limit a specific issue discussion to a single thread from now-on-wards. Thank you!

**Reply by appheadspace** [reply]
FYI. I didn't get the solution yet. So, Kindly consider my issue. @mcgratta @drjfloyd

**Reply by drjfloyd** [reply]
Look at the equation in the User's Guide for RAMP_T that I gave you in the other discussion. What is the independent variable T for RAMP_T for &RAMP? It is time. You are trying to use T as the x-coordinate. That is not what the Guide says RAMP_T does.

**Reply by appheadspace** [reply]
@drjfloyd In that case, Could you please tell me how to implement the desired temperature distribution to the wall in FDS? I need spatial distribution of temperature on wall. For eg: At x = 0.1m, temp = T1 , at x = 0.2m, temp = T2 ... etc. Thank you.

**Reply by drjfloyd** [reply]
Should you be specifying that temperature at all or should you just be assigning the actual wall materials from the experiment and let FDS solve the temperature?
If you feel you must specify those temperatures you can use multiple SURF each with its own TMP_FRONT.

**Reply by appheadspace** [reply]
@drjfloyd I want to specify the temperature directly (Not expecting FDS to solve it). Please see the approach-3 in my main question (i'm pasting it here below also). I think I've tried what you are telling me to do. It is not giving expected results. Please let me know if I'm wrong.



Approach -3:::

&OBST XB=0,0.750,-0.00005,0.00005,0,0 , SURF_ID='bottom_wall' /
&SURF ID = 'bottom_wall', COLOR = 'GRAY'/

&OBST XB =0,0,-0.00005,0.00005,0,0 ,SURF_ID='BW1'/
&SURF ID = 'BW1',TMP_FRONT =50/

&OBST XB =0.00015,0.00015,-0.00005,0.00005,0,0 ,SURF_ID='BW2'/
&SURF ID = 'BW2',TMP_FRONT =47.96/

&OBST XB =0.005025,0.005025,-0.00005,0.00005,0,0 ,SURF_ID='BW3'/
&SURF ID = 'BW3',TMP_FRONT =47.33/

bottom wall temperatures are showing 30 at all locations.
(PS: In my case INIT temp and TMPA , both are 30)

**Reply by drjfloyd** [reply]
What is the area of this OBST?

&OBST XB =0,0,-0.00005,0.00005,0,0 ,SURF_ID='BW1'/

**Reply by appheadspace** [reply]
.

**Reply by drjfloyd** [reply]
You did not answer my question. 

The User's Guide explains that the XB for an OBST will snap to the grid. Then every wall cell on that snapped OBST will get the SURF_ID.

What is the area of the OBST you defined?  Based on your grid and that area, how many wall cells will be assigned your SURF_ID?

**Reply by appheadspace** [reply, marked as answer]
XB =0,0,-0.00005,0.00005,0,0 is a line of length (0.00005x2 =) 0.0001 m  at x=0 and z = 0. Since line is 1D, so no area.

**Reply by drjfloyd** [reply]
Zero area means zero wall cells are being assigned your SURF_ID.

**Reply by appheadspace** [reply]
Oh! Thank you so much for explaining this. I will change my input file accordingly and try to get the result. Will come back here again if anything doesn't go right. Once again, thank you!
